extensions/blocks/calendly/calendly.php

```diff
@@ -86,35 +80,27 @@ function load_assets( $attr, $content ) {
 	);
 
 	if ( 'link' === $style ) {
-		wp_enqueue_style( 'jetpack-calendly-external-css', 'https://assets.calendly.com/assets/external/widget.css', null, JETPACK__VERSION );
-
-		/*
-		 * If we have some additional styles from the editor
-		 * (a custom text color, custom bg color, or both )
-		 * Let's add that CSS inline.
-		 */
-		if ( ! empty( $submit_button_text_color ) || ! empty( $submit_button_background_color ) ) {
-			$inline_styles = sprintf(
-				'#%1$s .wp-block-button__link{%2$s%3$s}',
-				esc_attr( $block_id ),
-				! empty( $submit_button_text_color )
-					? 'color:#' . sanitize_hex_color_no_hash( $submit_button_text_color ) . ';'
-					: '',
-				! empty( $submit_button_background_color )
-					? 'background-color:#' . sanitize_hex_color_no_hash( $submit_button_background_color ) . ';'
-					: ''
+		if ( ! wp_style_is( 'jetpack-calendly-external-css' ) ) {
+			wp_enqueue_style( 'jetpack-calendly-external-css', 'https://assets.calendly.com/assets/external/widget.css', null, JETPACK__VERSION );
+		}
+
+		// Render deprecated version of Calendly block if needed. New markup contains fallback link.
+		if ( false === strpos( $content, 'wp-block-jetpack-calendly__fallback' ) ) {
+			$content = deprecated_render_button_v1( $attr, $block_id, $classes, $url );
+		} else {
+			$content = preg_replace(
+				'/data-id-attr="placeholder"/',
+				'id="' . esc_attr( $block_id ) . '"',
+				$content
```

Comment:
I'd recommend to do a simpler thing here (again, inspired by what I did in #15918).

- Give a `uniqueId` attribute to the Button inner block, e.g. `uniqueId: 'calendly-widget-id'`.
- Do a simple string replace here: `$content = str_replace( 'calendly-widget-id', esc_attr( $block_id ), $content );`

(good call about the `esc_attr`! will need to update the Eventbrite PR! 🙂 )

Reply:
Thanks, I've simplified that as per your suggestion 🙂